libtransmission/rpc-server.cc

```diff
@@ -544,6 +588,108 @@ static auto constexpr ServerStartRetryDelayIncrement = int{ 5 };
 static auto constexpr ServerStartRetryDelayStep = int{ 3 };
 static auto constexpr ServerStartRetryMaxDelay = int{ 60 };
 
+static char const* tr_rpc_address_to_string(tr_rpc_address const* addr, char* buf, size_t buflen)
+{
+    TR_ASSERT(tr_rpc_address_is_valid(addr));
+
+    switch (addr->type)
+    {
+    case TR_RPC_AF_INET:
+        return evutil_inet_ntop(AF_INET, &addr->addr, buf, buflen);
+
+    case TR_RPC_AF_INET6:
+        return evutil_inet_ntop(AF_INET6, &addr->addr, buf, buflen);
+
+    case TR_RPC_AF_UNIX:
+        tr_strlcpy(buf, addr->addr.unixSocketPath, buflen);
+        return buf;
+
+    default:
+        return nullptr;
+    }
+}
+
+static std::string tr_rpc_address_with_port(tr_rpc_server const* server)
+{
+    tr_rpc_address const* addr = server->bindAddress;
+
+    char addr_buf[TrUnixAddrStrLen];
+    tr_rpc_address_to_string(addr, addr_buf, sizeof(addr_buf));
+
+    std::string addr_port_str{ addr_buf };
+    if (addr->type != TR_RPC_AF_UNIX)
+    {
+        addr_port_str.append(":" + std::to_string(tr_rpcGetPort(server)));
+    }
+    return addr_port_str;
+}
+
+static bool tr_rpc_address_from_string(tr_rpc_address* dst, std::string_view src)
+{
+    if (tr_strvStartsWith(src, TrUnixSocketPrefix))
+    {
+        if (src.length() >= TrUnixAddrStrLen)
```

Comment:
Not my first choice, but I see that @ckerr prefers to use `std::size(src)` instead.

Reply:
I'm not a C++ expert, but besides style/convention this code should be correct, right?

Reply:
:+1: on using `std::size(src)` as the default, but @mikedld is right that it's more of a styling choice akin to whitespace indentation, one's not intrinsically better than the other